### PR TITLE
[CARBONDATA-3548]Renamed index_handler to spatial_index property and indexColumn is changed to spatialColumn

### PR DESCRIPTION
**Pull request by VenuReddy2103**
### Why is this PR needed?
 Current code base has many types of indexes. To avoid the confusion and be more specific, have changed the index_handler property to spatial_index. Also changed the isIndexColumn/setIndexColumn to isSpatialColumn/setSpatialColumn respectively.
 
 ### What changes were proposed in this PR?
1. Have changed the index_handler property to spatial_index. Also changed the isIndexColumn/setIndexColumn to isSpatialColumn/setSpatialColumn respectively.
2. Documentation is updated accordingly.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2964/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1246/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2967/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1249/

File: geo/src/main/java/org/apache/carbondata/geo/GeoHashImpl.java [hunk 1]
```diff
@@ -86,54 +86,54 @@
 
 
   /**
-   * Initialize the geohash index handler instance.
```
Comment (jackylk) [inline geo/src/main/java/org/apache/carbondata/geo/GeoHashImpl.java:89]: please rename the class to `GeoHashIndex`

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java [hunk 2]
```diff
@@ -466,13 +466,13 @@ private CarbonCommonConstants() {
   public static final String INDEX_COLUMNS = "INDEX_COLUMNS";
 
   /**
-   * Index handler table property. It allows user to create a new sort column from the set of
+   * Spatial index table property. It allows user to create a new sort column from the set of
    * existing schema columns. And can generate value for the new column after parsing each row
-   * through custom handler.
+   * through custom index implementation.
```
Comment (jackylk) [inline core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java:471]: Please add more description or rephrase. From this description, I still not know how to use it, like what should be the string to fill in for this table property, and what data type can it accept, and what custom index is (you can use `{@link className}` to link it)

File: format/src/main/thrift/schema.thrift [hunk 3]
```diff
@@ -135,9 +135,9 @@ struct ColumnSchema{
 	17: optional list<ParentColumnTableRelation> parentColumnTableRelations;
 
   /**
-   * To specify if it is an index column. Its Default value is false
+   * To specify if it is an spatial column. Its Default value is false
 	 */
-	18: optional bool indexColumn;
+	18: optional bool spatialColumn;
```
Comment (jackylk) [inline format/src/main/thrift/schema.thrift:140]: Is this the `sourcecolumns` of the spatial index?
If yes
1. mention it in the description.
2. change name to `isSpatialSourceColumn`

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1297/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3015/

File: geo/src/main/java/org/apache/carbondata/geo/GeoHashImpl.java (hunk 1 above)
Comment (VenuReddy2103) [reply geo/src/main/java/org/apache/carbondata/geo/GeoHashImpl.java:89]: Renamed to GeoHashIndex

File: core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java (hunk 2 above)
Comment (VenuReddy2103) [reply core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java]: Have rephrased this description

File: format/src/main/thrift/schema.thrift (hunk 3 above)
Comment (VenuReddy2103) [reply format/src/main/thrift/schema.thrift:140]: This property is set to true for the spatial index column. Not for the source columns. Spatial index column is a non schema column created internally with SPATIAL_INDEX property. Source columns are normal table schema columns.

**Comment by kunal642** [comment]
@VenuReddy2103 Please rebase

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1323/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3042/

**Comment by VenuReddy2103** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3044/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1325/

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**